### ISSUE TITLE
RouterModule.forRoot() broken with AoT since 2.4.0

### ISSUE THREAD

**Issue by gallivantor** [angular/angular#13844]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```
(see https://stackoverflow.com/questions/41463860/how-to-use-routermodule-forroot-in-angular2-compiler-cli-ngc-command )

**Current behavior**
With Angular 2.4.x, running NGC to do an AOT compile results in the error:
Error: Error encountered resolving symbol values statically. Calling function 'RouterModule', function calls are not supported. Consider replacing the function or lambda with a reference to an exported function, resolving symbol AppRoutingModule in C:/git/mxt-frontend/landing-page-client/src/client/app/app-routing.module.ts, resolving symbol AppRoutingModule
**NOTE: Downgrading to 2.3.1 performs a successful AoT compilation on the same code**

**Expected behavior**
Code compiles succesfully

**Minimal reproduction of the problem with instructions**'
Assume you have an ngModule that uses RouterModule.forRoot such as this:
```
export const routes: Route[] = [
  {
    path: '',
    component: ApplicationlistComponent
  }
];

@NgModule( {
  imports: [ ApplicationsModule, RouterModule.forRoot( routes ) ],
  exports: [ RouterModule ],
  declarations: [ApplicationlistComponent]
} )
export class AppRoutingModule {
}

```

Run NGC to do an AoT compile of the app

**Please tell us about your environment:**
Windows 10

* **Angular version:** 2.4.2
* **Browser:** N/A 
* **Language:** Typescript 2.0.10
* **Node (for AoT issues):** 6.3.1

**Comment by DzmitryShylovich** [comment]
Pls add a small reproduction

**Comment by chuckjaz** [comment]
@gallivantor I tried to reproduce using the example provided with 2.4.2 and couldn't. Could you provide a github repository that I could clone?

**Comment by Jungers42** [comment]
I have also just gotten burned by this. Been attempting to introduce AoT into an existing application and hitting one challenge after another. Finally cleared away many others and I'm currently left with this exact problem.

Windows 10
- **Angular version**: 2.4.2
- **Browser**: N/A
- **Language**: Typescript 2.1.4
- **Node (for AoT issues)**: 6.9.1

**Comment by gallivantor** [comment]
This is strange. I created a cut-down version of my app to demonstrate the error, but having cut it down the error no longer happened.

Then, I tried upgrading my app to 2.4.2 again, and this time I didn't get the error.

So I'm a bit stumped by this, all I can say is that I can no longer reproduce it...
I was getting problems with Typescript 2.1.4 and downgraded it to 2.0.10 so perhaps that was actually what fixed this
